doc/internal/man3/OSSL_TIME.pod

```diff
@@ -0,0 +1,90 @@
+=pod
+
+=head1 NAME
+
+OSSL_TIME, OSSL_TIME_SECOND, OSSL_TIME_NEVER,
+ossl_time_now, ossl_time_time_to_timeval, ossl_time_compare,
+ossl_time_add, ossl_time_subtract
+- times and durations
+
+=head1 SYNOPSIS
+
+ #include "internal/time.h"
+
+ typedef OSSL_TIME;
+
+ #define OSSL_TIME_SECOND
+ #define OSSL_TIME_NEVER
+
+ OSSL_TIME ossl_time_now(void);
+ void ossl_time_time_to_timeval(OSSL_TIME t, struct timeval *out);
+
+ int ossl_time_compare(OSSL_TIME a, OSSL_TIME b);
+ OSSL_TIME ossl_time_add(OSSL_TIME a, OSSL_TIME b);
+ OSSL_TIME ossl_time_subtract(OSSL_TIME a, OSSL_TIME b);
+
+=head1 DESCRIPTION
+
+These functions allow the current time to be obtained and for basic
+arithmetic operations to be safely performed with times and durations.
+
```

Comment:
I think we should clarify that OSSL_TIME can represent a duration of time, or a point in time (representated as a duration relative to some contextually understood epoch).

We should probably also guarantee a minimum precision. I'm open about whether that precision is 1us or 1ns.

Reply:
`OSSL_TIME_SECOND` defines the precision, I'm not sure we need to be more specific than that.  Maybe if it were a public API...

I've added _It can represent both a specific point time or a duration._ to attempt to clarify the definition of `OSSL_TIME`.

Reply:
`OSSL_TIME_SECOND` defines the precision.  I think leaving the minimum unspecified is sufficient for now.  If this API goes public, we might want to reconsider.